### PR TITLE
Refactoring IgxGrid Paging tests (master)

### LINKED ISSUE #6958: Tests Refactoring: IgxGrid - Grid Paging
Guidelines for test refactoring: https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular

Related to #6438

### PR DESCRIPTION
**Pull request by HristoP96**
Closes #6958

### CODE REVIEW

File: projects/igniteui-angular/src/lib/test-utils/grid-functions.spec.ts [hunk 1]
```diff
@@ -1780,6 +1781,78 @@ export class GridFunctions {
         return headerTitle.parent;
     }
 
+    public static verifyGridPager(fix, rowsCount, firstCellValue, pagerText, buttonsVisibility) {
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/test-utils/grid-functions.spec.ts]: This method should be in the file grid.pagination.spec.ts. It is used only there and for a specific grid definition

File: projects/igniteui-angular/src/lib/test-utils/grid-functions.spec.ts [hunk 2]
```diff
@@ -1780,6 +1781,78 @@ export class GridFunctions {
...
+            const pagingButtons = GridFunctions.getPagingButtons(grid.nativeElement);
+            expect(pagingButtons.length).toEqual(4);
+            expect(pagingButtons[0].className.includes(BUTTON_DISABLED_CLASS)).toBe(buttonsVisibility[0]);
+            expect(pagingButtons[1].className.includes(BUTTON_DISABLED_CLASS)).toBe(buttonsVisibility[1]);
+            expect(pagingButtons[2].className.includes(BUTTON_DISABLED_CLASS)).toBe(buttonsVisibility[2]);
+            expect(pagingButtons[3].className.includes(BUTTON_DISABLED_CLASS)).toBe(buttonsVisibility[3]);
+        }
+    }
+
+    public static testPagingAPI(fix, grid: IgxGridComponent, page: (page: number) => void, desiredPageIndex: number = 2) {
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/test-utils/grid-functions.spec.ts]: This method should be in the file grid.pagination.spec.ts.

File: projects/igniteui-angular/src/lib/grids/grid/grid.pagination.spec.ts [hunk 3]
```diff
@@ -1,557 +1,435 @@
...
+            fix.detectChanges();
+            grid = fix.componentInstance.grid;
+        }));
 
-        // Go to next page
-        pagingButtons[2].dispatchEvent(new Event('click'));
-        tick();
-        fix.detectChanges();
-        verifyGridPager(fix, 3, '4', '2\xA0of\xA04', [false, false, false, false]);
+        afterAll(fakeAsync(() => {
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/grid/grid.pagination.spec.ts]: You can remove this. This is controlled by the configureTestSuite.

File: projects/igniteui-angular/src/lib/grids/grid/grid.pagination.spec.ts [hunk 4]
```diff
@@ -1,557 +1,435 @@
...
+        });
     });
 
-    it('"paginate" method should paginate correctly', fakeAsync(() => {
-        const fix = TestBed.createComponent(PagingComponent);
-        fix.detectChanges();
-        const grid = fix.componentInstance.grid;
-        testPagingAPI(fix, grid, (pageIndex) => grid.paginate(pageIndex));
-    }));
+    it('change paging with button', () => {
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/grid/grid.pagination.spec.ts]: This test can be make without clicking on buttons and just direct calling of the methods and this is done in the test 'should paginate data API', so you can remove the test

File: projects/igniteui-angular/src/lib/grids/grid/grid.pagination.spec.ts (hunk 3 above)
Comment (HristoP96) [reply projects/igniteui-angular/src/lib/grids/grid/grid.pagination.spec.ts]: Done

File: projects/igniteui-angular/src/lib/grids/grid/grid.pagination.spec.ts (hunk 4 above)
Comment (HristoP96) [reply projects/igniteui-angular/src/lib/grids/grid/grid.pagination.spec.ts]: Done

File: projects/igniteui-angular/src/lib/test-utils/grid-functions.spec.ts (hunk 2 above)
Comment (HristoP96) [reply projects/igniteui-angular/src/lib/test-utils/grid-functions.spec.ts]: Done

File: projects/igniteui-angular/src/lib/test-utils/grid-functions.spec.ts (hunk 1 above)
Comment (HristoP96) [reply projects/igniteui-angular/src/lib/test-utils/grid-functions.spec.ts]: Done

**Pull request merged**